starter_code/express-spotify-app/app.js

```diff
@@ -0,0 +1,71 @@
+const SpotifyWebApi   = require('spotify-web-api-node');
+const spotify         = new SpotifyWebApi();
+const express = require('express');
+const app = express();
+const expressLayouts = require('express-ejs-layouts');
+
+app.set('views', __dirname + '/views');
+app.set('view engine', 'ejs');
+app.set('layout', 'layouts/main-layout');
+
+
+app.use(express.static('public'));
+app.use(expressLayouts);
+
+
+// Remember to paste here your credentials
+const clientId = '8f1e69c9802e435e997e1983ec9348d0',
+    clientSecret = '4037307a11fd4f2ebc93931ea9b99e25';
+
+const spotifyApi = new SpotifyWebApi({
+  clientId : clientId,
+  clientSecret : clientSecret
+});
+
+// Retrieve an access token.
+spotifyApi.clientCredentialsGrant()
```

Comment:
This piece of code is asynchronous, so you can have problems if the response is not ready when the app needs it. You can try to wrap it in a function with a callback.